test/cypress/integration/users-management.spec.js

```diff
@@ -0,0 +1,169 @@
+
+  beforeEach(() => {
+    cy.appBuilderLogin();
+    cy.openPageFromMenu(['Users', 'Management'], true);
+  });
+
+  afterEach(() => {
+    cy.appBuilderLogout();
+  });
+
+  describe('Add a new user ', () => {
+    it('Should create a new user', () => {
+      const password = generateRandomId();
+      cy.log('Create a new user');
+      cy.getByTestId('UserListPage__AddUserButton').click();
+      cy.getByTestId('UserForm__UsernameField').type(username);
+      cy.getByTestId('UserForm__PasswordField').type(password);
+      cy.getByTestId('UserForm__PasswordConfirmField').type(password);
+      cy.getByTestId('UserForm__ProfileTypeField').select(PROFILE_TYPE_CODE);
+      cy.getByTestId('UserForm__SaveButton').click();
+      cy.log('Validate user creation');
+      cy.getTableRowsBySelector(username).should('be.visible');
+    });
+  });
+
+  describe('Validate user management page components', () => {
+    it('Validate user management page components', () => {
+      cy.validateBreadcrumbItems(['Users', 'Management']);
+      cy.get('h1').contains('Users');
+      cy.getByTestId('UserSearchForm_Form').should('be.visible').within(() => {
+        cy.get('h3').contains('Search');
+      });
+      cy.getByTestId('UserSearchForm__UsernameField').should('be.visible');
+      cy.getByTestId('UserSearchForm__WithProfileField').should('be.visible');
+      cy.getByTestId('UserSearchForm__SearchButton').should('be.visible');
+      cy.get('table').should('be.visible');
+      cy.getByTestId('UserListPage__AddUserButton').should('be.visible');
+    });
+  });
+
+  describe('Validate user management edit profile page components', () => {
+    it('Validate user management edit profile page components', () => {
+      cy.log('Validate user management edit profile page components');
+      cy.getTableActions(username).click();
+      cy.getVisibleActionItemByClass('UserListMenuAction__menu-item-edit-profile').click();
+      cy.validateUrlChanged(`/userprofile/${username}`);
+      cy.validateBreadcrumbItems(['Users', 'Management', 'Edit user profile']);
+      cy.getByTestId('UserProfileForm__UsernameField').should('be.visible')
+        .and('be.disabled');
+      cy.getByTestId('UserProfileForm__fullnameField').should('be.visible');
+      cy.getByTestId('UserProfileForm__emailField').should('be.visible');
+      cy.getByTestId('UserProfileForm__ProfileTypeField').should('be.visible');
+    });
+  });
+
+  describe('Validate user management view profile page components', () => {
+    it('Validate user management view profile page components', () => {
+      cy.log('Validate user management view profile page components');
+      cy.getTableActions(username).click();
+      cy.getVisibleActionItemByClass('UserListMenuAction__menu-item-view-profile').click();
+      cy.validateUrlChanged(`/user/view/${username}`);
+      cy.validateBreadcrumbItems(['Users', 'Management', 'Details']);
+      cy.get('table').contains('Username').should('be.visible');
```

Comment:
We should avoid selecting by HTML element. is there no other option to retrieve this table?